src/plugins/syscalls/syscalls.cpp

```diff
@@ -299,6 +328,45 @@ static void print_footer(output_format_t format, uint32_t nargs)
     }
 }
 
+static event_response_t win_ret_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    syscall_wrapper_t* wrapper = (syscall_wrapper_t*)info->trap->data;
+    syscalls* s = wrapper->sc;
+    int nargs = wrapper->wsc->num_args;
+    uint32_t thread_id = 0;
+
+    if ( !s || !wrapper->args )
+        goto exit;
+
+    if ( info->regs->cr3 != wrapper->target_cr3 )
+    {
+        PRINT_DEBUG("Syscall return callback hit but CR3 (0x%lx) doesn't match target process (0x%lx)\n",
+                    info->regs->cr3, wrapper->target_cr3);
+        goto exit;
```

Comment:
At least on Linux, I believe the correct thing to do here would be to single step the original instruction. Most syscalls are invoked via libc, which is a shared library that shares its pages across multiple processes. Thus, removing the breakpoint here would remove it for all processes since they all use the same underlying physical page. I don't know whether DRAKVUF or libvmi have the smarts to handle this complexity for you.

I'm pretty sure it's the same on Windows for the typical case.